fe/fe-core/src/main/java/org/apache/doris/planner/RuntimeFilterGenerator.java

```diff
@@ -122,6 +124,33 @@ private RuntimeFilterGenerator(Analyzer analyzer) {
         bloomFilterSizeLimits = new FilterSizeLimits(sessionVariable);
     }
 
+    private void collectAllTupleIdsHavingConjunct(PlanNode node, HashSet<TupleId> tupleIds) {
+        // for simplicity, skip join node( which contains more than 1 tuple id )
+        // we only look for the node meets either of the 2 conditions:
+        // 1. The node itself has conjunct
+        // 2. it's an aggregation or exchange node with no conjunct, but its descendant have conjuncts.
+        int tupleNumBeforeCheckingChildren = tupleIds.size();
+        for (PlanNode child : node.getChildren()) {
+            collectAllTupleIdsHavingConjunct(child, tupleIds);
+        }
+        if ((node.getTupleIds().size() == 1 && !node.conjuncts.isEmpty())) {
+            // The node itself has conjunct
+            tupleIds.add(node.getTupleIds().get(0));
+        } else if ((node instanceof AggregationNode || node instanceof ExchangeNode) && tupleIds.size() > tupleNumBeforeCheckingChildren) {
```

Comment:
Why only add aggregation or exchange node here, whether other nodes can also generate runtime filter if child has conjunct is satisfied.

Reply:
Yes I agree. This is not a perfect solution to only check agg and exchange node's children' conjunct. I added this code because after testing all tpch 22 queries, I found one useful runtime filter of q2 was removed. The goal of this modification is trying to keep all possible useful runtime filters and remove ones which are absolutely useless. ( that means we might still have some invalid ones in some cases ). For other node types, I am not so sure if this logic is still valid. So I think we can add more node types here if we come across new scenarios in future.

Reply:
I think there is no need to limit node types, including join nodes, which will make this pr less tricky and easier to understand and use.

You can try to remove `node.getTupleIds().size() == 1` and `node instanceof AggregationNode || node instanceof ExchangeNode`, and then test it on SSB and TPCH

In addition to agg and exchange nodes, the nodes that satisfy the number of tuples equal to 1 and the children have conjunct are mainly: sort, union, topn, table function, it seems that they can be generated.

Reply:
If don't need to limit the node type, this strategy can be placed after the entire runtime filter is generated, and you only need to traverse `runtimeFiltersByTid`.

Reply:
I tested on TPCH queries, only join node can be skipped when collecting the tuple ids of nodes which having conjunct. currently, join node has more then 1 descendant tuple ids, it's safe to skip it. And the first version is to remove the invalid runtime filters after entire runtime filter are generated, do you mean it's better to change it back to old way?